app/src/main/java/com/kickstarter/viewmodels/ActivityFeedViewModel.java

```diff
@@ -102,19 +107,32 @@ public ViewModel(final @NonNull Environment environment) {
 
       final Observable<Void> refreshSurvey = Observable.merge(this.refresh, this.resume).share();
 
-      this.currentUser.loggedInUser()
+      final Observable<User> loggedInUser = this.currentUser.loggedInUser();
+
+      loggedInUser
         .compose(takeWhen(refreshSurvey))
-        .switchMap(__ -> this.client.fetchUnansweredSurveys().compose(neverError()).share())
+        .switchMap(__ -> this.apiClient.fetchUnansweredSurveys().compose(neverError()).share())
         .compose(bindToLifecycle())
         .subscribe(this.surveys);
 
+      loggedInUser
+        .compose(takeWhen(refreshSurvey))
+        .map(User::unseenActivityCount)
+        .map(IntegerUtils::intValueOrZero)
+        .filter(IntegerUtils::isNonZero)
+        .distinctUntilChanged()
+        .switchMap(__ -> this.apolloClient.clearUnseenActivity().compose(neverError()))
+        .switchMap(__ -> this.apiClient.fetchCurrentUser().compose(neverError()))
+        .compose(bindToLifecycle())
+        .subscribe(freshUser -> this.currentUser.refresh(freshUser));
```

Comment:
So just to confirm I am reading this correctly:

1. User refreshes feed.
2. Take the current user and inspect their `unseenActivityCount`.
3. If this is _not_ zero, clear the activity count and refresh the user.

I guess the only edge-case race condition is if, between the user logging in and looking at the feed, new activities come in. Would they be alerted to this by the OS via the push notification in some way? I'm also kind've wondering how to handle this on iOS 🤔

Reply:
Yeah that's how it works. And hmm, they would get the push but no, we don't refresh the users on pushes. Should I maybe do a `combineLatest` like
```
refreshSurvey
. combineLatest(currentUser)
.filter(ObjectUtils::isNotNull)
//do the check and refresh
```

Reply:
Ooh also I think the situation with log in would be.
They're logged out. 
They log in.
`loggedInUser` emits.
They return to that screen. `onResume` is fired (a lifecycle method).
So we'd have the updated user.

But that only handles log in, not getting pushes or new activity while on that screen.

Reply:
I think if a push comes in while the user is in the app we should refresh the count, but if the user is on the activities view while that happens then I guess we can ignore it, seeing as we'd have to immediately clear it again.

Reply:
Spoke to @nnekab! She said for now, Android doesn't need to update the count after a push comes in because you can't actually see the number on that screen. It's only in the drawer in Discovery.